clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -541,4 +548,16 @@ private static void addPartitionData(String dest, Queue<Send> sends, Struct part
     public boolean shouldClientThrottle(short version) {
         return version >= 8;
     }
+
+    /*
+     * The buffer underlying a FetchResponse is sliced into many CompletedFetch objects.
```

Comment:
Kind of a meta comment, but this kind of bookkeeping seems like it would be better left inside `Fetcher`. This object is also used by the broker.